### PR TITLE
Sdit 714 upgrade hmpps sqs to spring cloud aws release candidate

### PR DESCRIPTION
**Pull request by steverendell**
formatting in separate commit

### CODE REVIEW

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt [hunk 1]
```diff
@@ -19,7 +19,7 @@ import uk.gov.justice.hmpps.sqs.HmppsSqsProperties.QueueConfig
 class HmppsQueueFactory(
   private val context: ConfigurableApplicationContext,
   private val healthContributorRegistry: HmppsHealthContributorRegistry,
-  private val sqsClientFactory: SqsClientFactory,
+  private val sqsClientFactory: SqsClientFactory
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt:22]: this looks the wrong way round - comma shouldn't have been removed here

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt [hunk 2]
```diff
@@ -77,7 +79,7 @@ class HmppsQueueFactory(
         override fun handle(message: org.springframework.messaging.Message<Any>, t: Throwable) {
           // SDI-477 remove this logging when we are comfortable that all is working as expected - instant retries
           log.info("Setting visibility of messageId ${message.headers["id"]} to $errorVisibilityTimeout (to initiate faster retry) after receiving exception ${t.cause?.cause?.cause}")
-          val sqsVisibility = message.headers["Sqs_Visibility"] as QueueMessageVisibility
+          val sqsVisibility = message.headers["Sqs_VisibilityTimeout"] as QueueMessageVisibility
```
Comment (petergphillips) [inline hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt:82]: what's going on here?

Comment (steverendell) [reply hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt:82]: It appears the name of the property has changed.

File: hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt (hunk 1 above)
Comment (steverendell) [reply hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt:22]: argh yes - thought I checked this !

**Review by petergphillips — APPROVED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**